app/src/main/java/eu/kanade/tachiyomi/ui/player/settings/PlayerSettingsScreenModel.kt

```diff
@@ -0,0 +1,149 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Info
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Switch
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import cafe.adriel.voyager.core.model.ScreenModel
+import eu.kanade.presentation.util.collectAsState
+import eu.kanade.tachiyomi.R
+import eu.kanade.tachiyomi.ui.player.settings.sheets.subtitle.toHexString
+import eu.kanade.tachiyomi.util.preference.toggle
+import `is`.xyz.mpv.MPVLib
+import tachiyomi.core.preference.Preference
+import tachiyomi.presentation.core.components.material.padding
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.io.File
+import java.io.InputStream
+
+val sheetDialogPadding = PaddingValues(vertical = MaterialTheme.padding.small, horizontal = MaterialTheme.padding.medium)
+
+class PlayerSettingsScreenModel(
+    val preferences: PlayerPreferences = Injekt.get(),
+    private val hasSubTracks: Boolean = true,
+) : ScreenModel {
+
+    fun togglePreference(preference: (PlayerPreferences) -> Preference<Boolean>) {
+        preference(preferences).toggle()
+    }
+
+    @Composable
+    fun ToggleableRow(
+        @StringRes textRes: Int,
+        paddingValues: PaddingValues = sheetDialogPadding,
+        isChecked: Boolean,
+        onClick: () -> Unit,
+        coloredText: Boolean = false,
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .clickable(onClick = onClick)
+                .padding(paddingValues),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Text(
+                text = stringResource(id = textRes),
+                color = if (coloredText) MaterialTheme.colorScheme.primary else Color.Unspecified,
+                style = MaterialTheme.typography.titleSmall,
+            )
+            Switch(
+                checked = isChecked,
+                onCheckedChange = null,
+            )
+        }
+    }
+
+    @Composable
+    fun OverrideSubtitlesSwitch(
+        content: @Composable () -> Unit,
+    ) {
+        val overrideSubtitles by preferences.overrideSubtitlesStyle().collectAsState()
+
+        val updateOverride = {
+            val overrideType = if (overrideSubtitles) "no" else "force"
+            togglePreference(PlayerPreferences::overrideSubtitlesStyle)
+            MPVLib.setPropertyString("sub-ass-override", overrideType)
```

Comment:
I think we should let the user change subtitle styles *without* forcing them to override ASS subtitles.
ASS subs can look quite bad when you override their styling, whereas typical WebVTT subtitles can be styled by the user exactly to their liking.

Reply:
Maybe the toggle should read "Override ASS subtitle styles" and be below the sub settings, while the sub settings are always enabled and pre-filled with the default values.
Maybe there should also be a button to "Reset to default style".